spring-cloud-gcp-starters/spring-cloud-gcp-starter-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/autoconfig/GcpPubSubAutoConfiguration.java

```diff
@@ -178,4 +181,16 @@ public SubscriptionAdminClient subscriptionAdminClient() {
 					ioe);
 		}
 	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	public SubscriptionAdminSettings subscriptionAdminSettings() {
+		try {
+			return SubscriptionAdminSettings.newBuilder().build();
```

Comment:
Looks like this is useless bean, without any configuration properties based customization.

So, maybe we should consider to use this expression in the `PubsubTemplate` directly:
```
private SubscriptionAdminSettings pullSettings = SubscriptionAdminSettings.newBuilder().build();
```
as the default value?

Reply:
The idea is that the user can provide their own `SusbcriptionAdminSettings` bean with their desired configuration, overriding the default one we provide.

Thoughts?

Reply:
Well, I agree that we should be able to eat user-provided `SusbcriptionAdminSettings` (optional one) in the `pubSubTemplate`, but since this bean is really void, I would prefer do not have it at all since it is going to be used in most cases. So, would be better to move it into the `PubSubTemplate` and create it on demand on first `pull` call if it isn't provided.